session.go

```diff
@@ -91,23 +92,75 @@ func NewSession(stdout, stderr io.Writer) (*Session, error) {
 	return s, nil
 }
 
+func (s *Session) initGoMod() (err error) {
+	hasMod, replaces, err := getModReplaces()
+	if err != nil || !hasMod {
+		return
+	}
+
+	tempModule := filepath.Base(s.tempDir)
+	goModPath := filepath.Join(s.tempDir, "go.mod")
+
+	mod := "module " + tempModule + "\n" + strings.Join(replaces, "\n")
+
+	return ioutil.WriteFile(goModPath, []byte(mod), 0644)
+}
+
+func getModReplaces() (hasMod bool, replaces []string, err error) {
+	pwd, err := os.Getwd()
+	if err != nil {
+		return
+	}
+
+	file, err := os.Open(pwd + "/go.mod")
+	if err != nil {
+		if os.IsNotExist(err) {
+			err = nil
+		}
+		return
+	}
+	defer file.Close()
+
+	out, err := exec.Command("go", "list", "-m", "all").Output()
+	s := bufio.NewScanner(bytes.NewReader(out))
+
+	s.Scan()
+	module := s.Text()
+	if module == "" {
+		return
+	}
+
+	hasMod = true
+	replaces = append(replaces, "replace "+module+" => "+pwd)
+
+	for s.Scan() {
+		replace := s.Text()
+		if strings.Contains(replace, "=>") {
+			replaces = append(replaces, "replace "+replace)
+		}
+	}
+
+	return
+}
+
 func (s *Session) init() (err error) {
 	s.fset = token.NewFileSet()
-	s.types = &types.Config{Importer: importer.Default()}
```

Comment:
@lvht 
the test is failed. I have trying to resolve the issues. please rollback this line. it would make test to pass. hope it would be help for you

Reply:
Because it is set type checking. if delete this line. error will happens at `commands.go:230`
```go
_, err = s.types.Check("_tmp", s.fset, []*ast.File{s.file}, &s.typeInfo)
  if err != nil {
    fmt.Println("error")
    debugf("typecheck error (ignored): %s", err)
}
```